tests/results/test_result_store.py

```diff
@@ -440,6 +441,55 @@ def bar():
 
     assert persist_result is True
 
+    with temporary_settings({PREFECT_TASKS_DEFAULT_PERSIST_RESULT: True}):
+        persist_result = bar()
+
+    assert persist_result is True
+
+
+async def test_task_can_opt_out_of_result_persistence_with_setting():
+    with temporary_settings({PREFECT_TASKS_DEFAULT_PERSIST_RESULT: True}):
+
+        @task(persist_result=False)
+        def bar():
+            return should_persist_result()
+
+        persist_result = bar()
+        assert persist_result is False
+
+        async def abar():
+            return should_persist_result()
+
+        persist_result = await abar()
+        assert persist_result is False
+
+
+async def test_can_opt_out_of_result_persistence_with_setting_when_flow_uses_feature():
+    with temporary_settings({PREFECT_TASKS_DEFAULT_PERSIST_RESULT: False}):
+
+        @flow(persist_result=True)
+        def foo():
+            return bar()
+
+        @task
+        def bar():
+            return should_persist_result()
+
+        persist_result = foo()
+
+        assert persist_result is False
```

Comment:
is the value of `should_persist_result` actually equivalent to whether the engine will persist the result? the conditional logic makes me think otherwise